File: java/rlib/BinaryTreeNode.java
package rlib;

public class BinaryTreeNode<T> {

  public T data;
  public BinaryTreeNode<T> left, right, parent;

  public BinaryTreeNode() {}

  public BinaryTreeNode(T data) {
    this.data = data;
  }

  public BinaryTreeNode(
    T data,
    BinaryTreeNode<T> left,
    BinaryTreeNode<T> right)
  {
    this.data = data;
    this.left = left;
    this.right = right;
  }

  public BinaryTreeNode(
    T data,
    BinaryTreeNode<T> left,
    BinaryTreeNode<T> right,
    BinaryTreeNode<T> parent)
  {
    this.data = data;
    this.left = left;
    this.right = right;
    this.parent = parent;
  }

}

File: python/rlib/binary_tree_node.py
class BinaryTreeNode:
    
    def __init__(self, data = None, left = None, right = None, parent = None):
        self.data = data
        self.left = left
        self.right = right
        self.parent = parent

if __name__ == '__main__':
    root = BinaryTreeNode(1)
    print root